MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,72 +1,120 @@
+    private var currentQuestion: QuizQuestion?
+    private var alertPresenter: AlertPresenterProtocol?
+    private var statisticService: StatisticService = StatisticServiceImplementation()
+    
+    private func convert(model: QuizQuestion) -> QuizStepViewModel {
+        // преобразовываем данные модели вопроса в те, что нужно показать на экране
+        return QuizStepViewModel(
+            image: UIImage(named: model.image) ?? UIImage(),
+            question: model.text,
+            questionNumber: "\(currentQuestionIndex + 1)/\(questionsAmount)")
+    }
+    
+    private func showQuestion(quiz step: QuizStepViewModel) {
+        // здесь мы заполняем нашу картинку, текст и счётчик данными
+        imageView.image = step.image
+        textLabel.text = step.question
+        counterLabel.text = step.questionNumber
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        alertPresenter = AlertPresenter(viewController: self)
+        questionFactory = QuestionFactory(delegate: self)
+        questionFactory?.requestNextQuestion()
+    }
+    
+    // MARK: - QuestionFactoryDelegate
+    func didRecieveNextQuestion(question: QuizQuestion?) {
+        guard let question = question else {
+            return
+        }
+        currentQuestion = question
+        let viewModel = convert(model: question)
+        DispatchQueue.main.async { [weak self] in
+            self?.showQuestion(quiz: viewModel)
+        }
+    }
+    
+    private func showAnswerResult(isCorrect: Bool) {
+        // отображаем результат ответа (выделяем рамкой верный или неверный ответ)
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        imageView.layer.cornerRadius = 20
+        imageView.layer.borderColor = isCorrect ? UIColor.ypGreen?.cgColor : UIColor.ypRed?.cgColor
+        noButton.isEnabled = false
+        yesButton.isEnabled = false
+        if isCorrect {
+            correctAnswers += 1
+        }
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [weak self] in
+            guard let self = self else { return }
+            self.imageView.layer.borderWidth = 0
+            self.noButton.isEnabled = true
+            self.yesButton.isEnabled = true
+            self.showQuestionOrResult()
+        }
+    }
+    
+    private func showQuestionOrResult() {
+        
+        if currentQuestionIndex == questionsAmount - 1 { // - 1 потому что индекс начинается с 0, а длинна массива — с 1
+            // показать результат квизa
+            statisticService.store(correct: correctAnswers, total: questionsAmount)
+            let textMessage =
+            """
+Ваш результат: \(correctAnswers) из 10,
+Количество сыгранных квизов: \(statisticService.gamesCount)
+Рекорд: \(statisticService.bestGame.correct)/\(statisticService.bestGame.total) (\(statisticService.bestGame.date.dateTimeString))
+Средняя точность: \(String(format: "%.2f", statisticService.totalAccuracy))%
+"""
```

Comment:
Код гораздо проще читать, когда вложенность "столбцов" одинаковая.
```suggestion
            Ваш результат: \(correctAnswers) из 10,
            Количество сыгранных квизов: \(statisticService.gamesCount)
            Рекорд: \(statisticService.bestGame.correct)/\(statisticService.bestGame.total) (\(statisticService.bestGame.date.dateTimeString))
            Средняя точность: \(String(format: "%.2f", statisticService.totalAccuracy))%
            """
```